### PR TITLE
convert a string to a raw binary string of 0s and 1s

### PR DESCRIPTION
**Pull request by fdncred**
# Description

This PR converts a string into a raw binary represented by a string of 0s and 1s padded to 8 digits with zeros.

This is useful for encoding data.
image

# User-Facing Changes

# Tests + Formatting

# After Submitting

### CODE REVIEW

File: crates/nu-command/src/conversions/into/binary.rs [hunk 1]
```diff
@@ -25,13 +35,15 @@ impl Command for SubCommand {
...
                 (Type::Date, Type::Binary),
+                (Type::String, Type::String),
             ])
             .allow_variants_without_examples(true) // TODO: supply exhaustive examples
             .rest(
                 "rest",
                 SyntaxShape::CellPath,
                 "for a data structure input, convert data at the given cell paths",
             )
+            .switch("raw", "Convert to raw", Some('r'))
```
Comment (amtoine) [inline crates/nu-command/src/conversions/into/binary.rs:46]: as this only applies to `string`s, maybe we should mention it in the description of the option? e.g.
> Convert strings to raw binary. Do not apply on other datatypes.

**Review by amtoine — CHANGES_REQUESTED** [review]

File: crates/nu-command/src/conversions/into/binary.rs (hunk 1 above)
Comment (fdncred) [reply crates/nu-command/src/conversions/into/binary.rs]: thanks!

**Comment by sholderbach** [comment]
I am honestly against placing this functionality inside the `into binary` command. The `into ...` casting commands in my view should be true to the declared type in the name and only return that and use options sparingly to resolve ambiguities (like we have with `into datetime`). I think this functionality would be better placed among the banished `bits` or `bytes` commands.

**Comment by fdncred** [comment]
I looked at putting this in `into int` and i thought it fit better here because it returns binary, although a text representation. I thought about bits too but I'm not sure if `into bits` is right either.

**Comment by amtoine** [comment]
thanks @fdncred for the fixes :blush:
as for the location of this, i do not have a strong preference, just keep in mind that the `bits` and `bytes` are gonna go in the `extra` feature :+1:

**Pull request merged**